### ISSUE TITLE
Multiple calls of RemoteFileSystem.get() ineffieciently log in and out

### ISSUE THREAD

**Issue by citynorman** [spotify/luigi#2252]
Not sure if I am using `luigi.contrib.ftp` correctly but it's extremely slow to download multiple files because of repeated logins. There are way to use a shared connection between subsequent calls of `RemoteFileSystem.get()`?

Checked out [ftp_experiment_outputs.py](https://github.com/spotify/luigi/blob/master/examples/ftp_experiment_outputs.py) but that's just for one file

PS FTP sucks!

    from luigi.contrib.ftp import RemoteFileSystem
    import ftputil
    
    [...]
    
    # this repeated logs in and out
    ftpfs = RemoteFileSystem(cfg_ftp_host, cfg_ftp_usr, cfg_ftp_pwd)
    for file in files_ftp_sync:
        ftpfs.get(file, cfg_dir_local+file)

    # this uses a single login
    with ftputil.FTPHost(cfg_ftp_host, cfg_ftp_usr, cfg_ftp_pwd) as ftp_host:
        for file in files_ftp_sync:
            print('$$$',cfg_dir_local+file)
            ftp_host.download(file, cfg_dir_local+file)

    # this doesn't work
    with RemoteFileSystem(cfg_ftp_host, cfg_ftp_usr, cfg_ftp_pwd) as ftpfs:
        for file in files_ftp_sync:
            print('$$$',cfg_dir_local+file)
            ftpfs.get(file, cfg_dir_local+file)

*proftp log*
2017-10-12 01:32:05,756 ubuntu-xenial proftpd[2205] ubuntu-xenial (192.168.33.1[192.168.33.1]): FTP session opened.
2017-10-12 01:32:07,715 ubuntu-xenial proftpd[2205] ubuntu-xenial (192.168.33.1[192.168.33.1]): USER travelclick: Login successful.
2017-10-12 01:32:07,769 ubuntu-xenial proftpd[2205] ubuntu-xenial (192.168.33.1[192.168.33.1]): FTP session closed.
2017-10-12 01:32:07,829 ubuntu-xenial proftpd[2206] ubuntu-xenial (192.168.33.1[192.168.33.1]): FTP session opened.
2017-10-12 01:32:10,083 ubuntu-xenial proftpd[2206] ubuntu-xenial (192.168.33.1[192.168.33.1]): USER travelclick: Login successful.
2017-10-12 01:32:10,107 ubuntu-xenial proftpd[2206] ubuntu-xenial (192.168.33.1[192.168.33.1]): FTP session closed.
2017-10-12 01:32:10,202 ubuntu-xenial proftpd[2207] ubuntu-xenial (192.168.33.1[192.168.33.1]): FTP session opened.
2017-10-12 01:32:12,529 ubuntu-xenial proftpd[2207] ubuntu-xenial (192.168.33.1[192.168.33.1]): USER travelclick: Login successful.
2017-10-12 01:32:12,531 ubuntu-xenial proftpd[2207] ubuntu-xenial (192.168.33.1[192.168.33.1]): FTP session closed.

*setup*

ftputil==3.3.1
luigi==2.6.2

Python 3.6.1 |Anaconda custom (64-bit)| (default, May 11 2017, 13:09:58) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.

**Comment by dlstadther** [comment]
@citynorman I agree that the ftp module here is less than optimal. It has been a bit neglected and could really use a refactor. Please feel free to contribute back with better code! It would be much appreciated!
